src/Components/Pure-Functions/addToCart.jsx

```diff
@@ -0,0 +1,25 @@
+const FuncAddToCart = (state, action) => {
+  const productExist = state.productsInCart.find(
+    (items) => items._id === action.payload._id
+  );
+  return {
+    ...state,
+    totalPrice: state.totalPrice + action.payload.price,
+    discountPrice: state.discountPrice + action.payload.discardPrice,
+    productsInCart: !productExist
```

Comment:
Why not just switch the two statements inside ternary operator instead of setting NOT operand to your state variable?